### DISCUSSION TITLE
Weird errors using PiecewiseMultilinear

### DISCUSSION THREAD

**Discussion by heinono1** [idaholab/moose#22625] [Q&A General] [answered]
I am trying to read in a function that gives the time dependence of some parameters. I am using PiecewiseMultilinear to read in the function from a txt file, but I get a behavior and errors I do not understand. In the input file, I have the Functions block:
```
[Functions]
  []
    [./current_waveform]
       type = PiecewiseMultilinear
       data_file = test2.txt
  []
```
If test2.txt is the following, the code runs without errors:
```
AXIS T
0 0.01 0.02 0.03 0.04 0.05 0.06 0.07 0.08 0.09 0.1
DATA
1 1 1 1 1 1 1 1 1 1 1
```
But if test2.txt is the following I get errors:
```
AXIS T
0. 0.00001 0.00002 0.00003 0.00004 0.00005 0.00006 0.00007 0.00008 0.00009 0.0001 0.00011 0.00012 0.00013 0.00014 0.00015 0.00016
DATA
0. 20.1099 36.7926 50.6321 62.4198 74.4977 87.0543 100.55 105.464 94.3073 85.0521 77.3742 70.8347 64.1341 57.168 49.6811 46.9548
```
The errors are 
```
*** ERROR ***
No valid AXIS lines found by GriddedData


[node02:3012710] [[64356,0],0] ORTE_ERROR_LOG: Data unpack had inadequate space in file util/show_help.c at line 513

or

*** ERROR ***
No valid AXIS lines found by GriddedData

MPI_ABORT was invoked on rank 4 in communicator MPI_COMM_WORLD
with errorcode 1.

(without the "Data unpack had inadequate space....")
```
I have no clue what causes these errors - any help will be much appreciated!

--------------------------------------------------------------------------

**Comment by GiudGiud** [comment]
Does it not like the `0.` ?
I have no clue either. Wont have time in the near future to debug this

**Comment by lynnmunday** [comment]
I just tried your failing gridded data file with the input file:
/moose/test/tests/functions/piecewise_multilinear/time.i
And it worked fine.  I didn't get an error.

**Reply by heinono1** [reply]
Weird. I have to try to find if there are some strange hidden characters that are read in in my case.

**Reply by heinono1** [reply]
Indeed in looks like some hidden extra characters had crept into the file as I generated it from a spreadsheet. Never trust Excel!

**Reply by hugary1995** [reply]
`dos2unix` is your friend here.

**Comment by heinono1** [comment, marked as answer]
Weird. I have to try to find if there are some strange hidden characters that are read in in my case.

**Reply by aaelmeli** [reply]
I think you are working on WSL.  Try to copy the working test2.txt file using "cp test2.txt somename.txt" and modify the numbers. Do not copy the number from the not working file. I do think it is a formating issue.

**Reply by heinono1** [reply]
Yes, thanks. It was a formatting issue.
